internal/pkg/otel/extension/elasticdiagnostics/extension_test.go

```diff
@@ -65,14 +87,39 @@ func TestExtension(t *testing.T) {
 		// test that the response can be unmarshalled
 		require.NoErrorf(collect, json.Unmarshal(b, &res), "failed to unmarshal response: %s", string(b))
 		require.NotEmpty(collect, res.GlobalDiagnostics)
+		require.NotEmpty(collect, res.ComponentDiagnostics)
 		foundCPU := false
 		for _, global := range res.GlobalDiagnostics {
 			if global.Name == "cpu" {
 				foundCPU = true
 				break
 			}
+			if strings.HasSuffix(global.Filename, "profile.gz") {
+				verifyPprof(t, global.Content)
+			}
 		}
 		require.True(collect, foundCPU, "cpu.pprof not found in global diagnostics")
+
+		for _, comp := range res.ComponentDiagnostics {
+			switch comp.Name {
+			case "test_component":
+				require.Equal(collect, "test.json", comp.Filename)
+				require.Equal(collect, "application/json", comp.ContentType)
+				require.Equal(collect, "Test Hook", comp.Description)
+				require.Equal(collect, []byte("diagnostic data"), comp.Content)
+			case "test_component2":
+				require.Equal(collect, "test2.json", comp.Filename)
+				require.Equal(collect, "application/json", comp.ContentType)
+				require.Equal(collect, "Test Hook2", comp.Description)
+				require.Equal(collect, []byte("diagnostic data 2"), comp.Content)
+			}
+		}
 		require.NoError(collect, resp.Body.Close())
 	}, 10*time.Second, 1*time.Millisecond, "extension did not start in time")
 }
+
+func verifyPprof(t *testing.T, content []byte) {
+	prof, err := profile.ParseData(content)
```

Comment:
TIL this exists, though I think it might accept the non-gz compressed variants that `go tool pprof` doesn't like (just quickly skimming the docs)